src/main/java/Storage.java

```diff
@@ -0,0 +1,168 @@
+     * @throws Exception if file does not exist or loading task error
+     */
+    public ArrayList<Task> load() throws Exception {
+        try {
+            if (file.exists()) {
+                return loadTasks();
+            } else {
+                throw new DukeException("New Path: File does not exist");
+            }
+        } catch (Exception e) {
+            throw new DukeException(e.getMessage());
+        }
+    }
+
+    /**
+     * Helper Function to load() method.
+     *
+     * @return list of tasks from text file
+     * @throws IOException if task type != "T" || "D" || "E"
+     * */
+    public ArrayList<Task> loadTasks() throws IOException {
+        ArrayList<Task> tasks = new ArrayList<Task>();
+        try {
+            FileReader reader = new FileReader(file);
+            BufferedReader bufferedReader = new BufferedReader(reader);
+
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] arr = line.split(" \\| ");
+                Task task;
+
+                switch (arr[0]) {
+                case "T": //todo
+                    task = new Todo(arr[2]);
+                    break;
+                case "D": //deadline
+                    task = new Deadline(arr[2], arr[3]);
+                    break;
+                case "E": //event
+                    task = new Deadline(arr[2], arr[3]);
+                    break;
+                default:
+                    throw new AssertionError("LOAD FILE FAILURE: invalid task type " + arr[0]);
+                }
+
+                if (Integer.parseInt(arr[1]) == 1) {
+                    task.markAsDone();
+                }
+                tasks.add(task);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return tasks;
+    }
+
+    /**
+     * Adds new tasks information in text file.
+     *
+     * @param task Task to save in file
+     * @param type Type of task of one Char length
+     * */
+    public void saveTasks(Task task, String type) {
+        try {
+            FileWriter writer = new FileWriter(file, true); //initialize file writer
+            BufferedWriter bufferedWriter = new BufferedWriter(writer);
+
+            int status;
+            if (task.getStatus() == true) {
+                status = 1;
+            } else {
+                status = 0;
+            }
+            bufferedWriter.write(type + " | " + status + " | " + task.getDescription());
+            bufferedWriter.newLine();
+            bufferedWriter.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Edits existing tasks in file through delete or update
+     *
+     * @param task Task to save in file
+     * @param command String command from user input
+     * @param position Index of task in TaskList
+     * */
+    public void updateTasks(Task task, String command, int position) {
```

Comment:
Hi, so we were reviewing your code during our tutorial, and do you think it might be better if you abstract out certain portions of this method into separate methods? Otherwise it's really good stuff great job!